data_model/src/events/data/events.rs

```diff
@@ -5,35 +5,38 @@ use iroha_primitives::small::SmallVec;
 
 use super::*;
 
+/// Generic [`MetadataChanged`] struct.
+/// Depending on wrapping event could mean inserted or removed metadata `(key, value)` pair.
+#[derive(
+    Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Decode, Encode, Deserialize, Serialize, IntoSchema,
+)]
+#[allow(missing_docs)]
+pub struct MetadataChanged<ID> {
```

Comment:
```suggestion
pub struct MetadataChanged<ID: Identifiable> {
```

maybe?

Reply:
If the trait bound isn't necessary it's better to have the structure as generic as possible.

Reply:
Initially it is what i do, but such approach poorly work with derive macroses, because it require us to manually put trait bounds onto `<ID as Identifiable>::Id`.